wpilibj/src/main/java/edu/wpi/first/wpilibj/experimental/command/SelectCommand.java

```diff
@@ -0,0 +1,96 @@
+package edu.wpi.first.wpilibj.experimental.command;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Supplier;
+
+import edu.wpi.first.wpilibj.command.IllegalUseOfCommandException;
+
+/**
+ * Runs one of a selection of commands, either using a selector and a key->command mapping, or a
+ * supplier that returns the command directly at runtime.  Does not actually schedule the selected
+ * command - rather, the command is run through this command; this ensures that the command will
+ * behave as expected if used as part of a CommandGroup.  Requires the requirements of all included
+ * commands, again to ensure proper functioning when used in a CommandGroup.  If this is undesired,
+ * consider using {@link ScheduleCommand}.
+ *
+ * <p>As this command contains multiple component commands within it, it is technically a command
+ * group; the command instances that are passed to it cannot be added to any other groups, or
+ * scheduled individually.
+ *
+ * <p>As a rule, CommandGroups require the union of the requirements of their component commands.
+ *
+ * @param <K> the type of the selector to use, e.g. String or enum
```

Comment:
It wasn't serving much purpose; the thing works just as well with `Object`.  Additionally, it was mildly confusing when the constructor that takes a `Supplier<Command>` was used, since the user would still have to supply a type variable despite it not being used.

Reply:
I think we might want to revisit this idea because it would help the user enforce the correct type when putting values in.  They wouldn't need to specify a type but the intelisense would warn them that it is probably a good idea to do so.